api/src/main/java/keywhiz/api/model/SecretDeletionMode.java

```diff
@@ -0,0 +1,17 @@
+package keywhiz.api.model;
+
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+
+public final class SecretDeletionMode {
+  private SecretDeletionMode() {}
+
+  public static final String HARD = "hard";
+  public static final String SOFT = "soft";
+
+  private static final List<String> values = ImmutableList.of(HARD, SOFT);
+
+  public static List<String> values() {
+    return values;
+  }
+}
```

Comment:
This class is intended to be used by clients. Generally speaking, Java enums don't play well with API evolution (unknown values result in exceptions), so we prefer strings. However, it's still nice to have a constants class available.